Libraries/LibJS/Runtime/JSONObject.cpp

```diff
@@ -347,11 +347,11 @@ ThrowCompletionOr<ByteString> JSONObject::serialize_json_array(VM& vm, Stringify
 
     state.seen_objects.remove(&object);
     state.indent = previous_indent;
-    return builder.to_byte_string();
+    return builder.to_string_without_validation();
 }
 
 // 25.5.2.2 QuoteJSONString ( value ), https://tc39.es/ecma262/#sec-quotejsonstring
-ByteString JSONObject::quote_json_string(ByteString string)
+String JSONObject::quote_json_string(String string)
```

Comment:
I think you can just take a `StringView` here instead (Or `Utf8View`). That would avoid this churn on line 242 above:
```c++
quote_json_string(MUST(String::from_byte_string(key.to_string()))),
```